qiskit/opflow/gradients/circuit_qfis/lin_comb_full.py

```diff
@@ -179,15 +203,42 @@ def convert(
 
                 # Compute −4 * Re(〈∂kψ|ψ〉〈ψ|∂lψ〉)
                 def phase_fix_combo_fn(x):
-                    return 4 * (-0.5) * (x[0] * np.conjugate(x[1]) + x[1] * np.conjugate(x[0]))
-
-                phase_fix = ListOp(
-                    [phase_fix_states[i], phase_fix_states[j]], combo_fn=phase_fix_combo_fn
-                )
-                # Add the phase fix quantities to the entries of the QFI
-                # Get 4 * Re[〈∂kψ|∂lψ〉−〈∂kψ|ψ〉〈ψ|∂lψ〉]
-                qfi_ops += [SummedOp(qfi_op) + phase_fix]
+                    return -4 * np.real(x[0] * np.conjugate(x[1]))
+
+                if phase_fix:
+                    phase_fix_op = ListOp(
+                        [phase_fix_states[i], phase_fix_states[j]], combo_fn=phase_fix_combo_fn
+                    )
+                    # Add the phase fix quantities to the entries of the QFI
+                    # Get 4 * Re[〈∂kψ|∂lψ〉−〈∂kψ|ψ〉〈ψ|∂lψ〉]
+                    qfi_ops += [SummedOp(qfi_op) + phase_fix_op]
+                else:
+                    qfi_ops += [SummedOp(qfi_op)]
 
             qfi_operators.append(ListOp(qfi_ops))
-        # Return the full QFI
-        return ListOp(qfi_operators, combo_fn=triu_to_dense)
+
+        # Return estimate of the full QFI -- A QFI is by definition positive semi-definite.
+        return ListOp(
+            qfi_operators, combo_fn=lambda x: check_and_realpart_extraction(triu_to_dense(x))
```

Comment:
It seems the check_and_realpart function can raise an error, but this aspect is not included in the docstring. Can anything be said/recommended about how to increase number of shots. Would it be a good idea to include the magnitude of the error in the message so if they increase shots and get a similar error at least there is someway to see what effect that had.
Also I see ETOL is some class level constant - would it make sense to have some instance level parameter to allow a user to alter this where it defaults to  this class level value but does allow that to be changed?

Reply:
I agree on the former. 
For the ETOL, I believe that for now it is ok to have this set as a constant. Right now it's only important to check if we have unreasonably large imaginary values or if they arise from numerical instabilities. If we reach the point were we have to balance this off in a delicate way we can quickly fix it.
But for now, I believe that this choice could rather confuse the user.

Reply:
I agree it can always be made into some parameter later say with the current fixed value as its default. I will note that should an error be raised due to values outside of this then a user is going to be exposed one way to the concept - but maybe its less confusing if their corrective action is just to alter shots rather than have this tolerance to be able to play around with as well.